docs: improve doc (#1568)
NVIDIA/NeMo-Guardrails: docs/user-guides/community/gliner.md

# GLiNER Integration

> [GLiNER](https://github.com/urchade/GLiNER) is a generalist and lightweight model for named entity recognition. It can detect a wide range of entity types, including comprehensive PII (Personally Identifiable Information) categories.

Reviewer: ```suggestion
[GLiNER](https://github.com/urchade/GLiNER) is a generalist and lightweight model for named entity recognition. [NVIDIA GLiNER-PII](https://huggingface.co/nvidia/gliner-PII) is an adaptation of this base model that can detect a wide range of entity types, including comprehensive PII (Personally Identifiable Information) categories.
```

This integration enables the NeMo Guardrails library to use a GLiNER-compatible server for PII detection and masking in input, output, and retrieval flows.

## Server Setup

> 1. Deploy a GLiNER-compatible server.

Reviewer: ```suggestion
Deploy a GLiNER-compatible server.
```

Once configured, the GLiNER integration can automatically:

1. Detect or mask PII in user inputs before the LLM processes them.
2. Detect or mask PII in LLM outputs before sending them back to the user.
3. Detect or mask PII in retrieved chunks before sending them to the LLM.

## Example Deployment

The [`examples/deployment/gliner_server/`](../../../examples/deployment/gliner_server/) directory provides an example GLiNER server implementation.
This implementation:

> - Uses the `nvidia/gliner-PII` model for comprehensive PII detection.

Reviewer: ```suggestion
- Uses the [NVIDIA GLiNER-PII](https://huggingface.co/nvidia/gliner-PII) model for comprehensive PII detection.
```
